backend/kernelCI_app/views/hardwareDetailsView.py

```diff
@@ -153,79 +156,221 @@ class HardwareDetails(View):
     required_params_get = ["origin"]
     cache_key_get = "hardwareDetailsGET"
 
+    def __init__(self):
+        self.filterTestDurationMin, self.filterTestDurationMax = None, None
+        self.filterBootDurationMin, self.filterBootDurationMax = None, None
+        self.filterBuildDurationMin, self.filterBuildDurationMax = None, None
+        self.filterBootStatus = set()
+        self.filterTestStatus = set()
+        self.filterTreeDetailsConfigs = set()
+        self.filterTreeDetailsCompiler = set()
+        self.filterArchitecture = set()
+        self.filterTestPath = ""
+        self.filterBootPath = ""
+        self.filterValid = set()
+        self.filter_handlers = {
+            "boot.status": self.__handle_boot_status,
+            "boot.duration": self.__handle_boot_duration,
+            "test.status": self.__handle_test_status,
+            "test.duration": self.__handle_test_duration,
+            "duration": self.__handle_build_duration,
+            "config_name": self.__handle_config_name,
+            "compiler": self.__handle_compiler,
+            "valid": self.__handle_valid,
+            "architecture": self.__handle_architecture,
+            "test.path": self.__handle_path,
+            "boot.path": self.__handle_path,
+        }
+
+    def __handle_boot_status(self, current_filter):
+        self.filterBootStatus.add(current_filter["value"])
+
+    def __handle_boot_duration(self, current_filter):
+        value = current_filter["value"][0]
+        operation = current_filter["comparison_op"]
+        if operation == "lte":
+            self.filterBootDurationMax = toIntOrDefault(value, None)
+        else:
+            self.filterBootDurationMin = toIntOrDefault(value, None)
+
+    def __handle_test_status(self, current_filter):
```

Comment:
don't use `__` that was a missconception that we had, private methods have only one underscore before them

also, these handlers are very close to the one in hardware details view, se if you can unite them

the trick is to receive what they are modifying as a parameter
or to return the value
I'd rather have them returning the value